lib/models/ObjectMDAzureInfo.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Helper class to ease access to the Azure specific information for
+ * Blob and Container objects.
+ */
+class ObjectMDAzureInfo {
+    /**
+     * @constructor
+     * @param {object} obj - Raw structure for the Azure info on Blob/Container
+     * @param {string} obj.containerPublicAccess - Public access authorization
+     *   type
+     * @param {object[]} obj.containerStoredAccessPolicies - Access policies
+     *   for Shared Access Signature bearer
+     * @param {object} obj.containerImmutabilityPolicy - data immutability
+     *   policy for this container
+     * @param {boolean} obj.containerLegalHoldStatus - legal hold status for
+     *   this container
+     * @param {boolean} obj.containerDeletionInProgress - deletion in progress
+     *   indicator for this container
+     * @param {string} obj.blobType - defines the type of blob for this object
+     * @param {string} obj.blobContentMD5 - whole object MD5 sum set by the
+     *   client through the Azure API
+     * @param {object} obj.blobCopyInfo - information pertaining to past and
+     *   pending copy operation targeting this object
+     * @param {number} obj.blobSequenceNumber - sequence number for a PageBlob
+     * @param {Date} obj.blobAccessTierChangeTime - date of change of tier
+     * @param {boolean} obj.blobUncommitted - A block has been put for a
+     *   nonexistent blob which is about to be created
+     */
+    constructor(obj) {
+        this._data = {
+            containerPublicAccess: obj.containerPublicAccess,
+            containerStoredAccessPolicies: obj.containerStoredAccessPolicies,
```

Comment:
was also wondering if it would beneficial to validate the structure of these properties.

Reply:
it's not worth the complexity